### PR TITLE
api: add in/out/error type schema to v1alpha1.Endpoint

### PR DESCRIPTION
**Pull request by lburgazzoli**
**Release Note**
```release-note
NONE
```

### CODE REVIEW

**Review by oscerd — APPROVED** [review]

File: pkg/apis/camel/v1alpha1/kamelet_binding_types.go [hunk 1]
```diff
@@ -42,6 +42,8 @@ type Endpoint struct {
 	URI *string `json:"uri,omitempty"`
 	// Properties are a key value representation of endpoint properties
 	Properties EndpointProperties `json:"properties,omitempty"`
+	// Types defines the schema of the data pproduced/consumed by the endpoint
```
Comment (astefanutti) [inline pkg/apis/camel/v1alpha1/kamelet_binding_types.go:45]: `pproduced` -> `produced`

**Review by nicolaferraro — APPROVED** [review]

**Review by astefanutti — APPROVED** [review]
Should the CRD be regenerated?

File: pkg/apis/camel/v1alpha1/kamelet_binding_types.go (hunk 1 above)
Comment (lburgazzoli) [reply pkg/apis/camel/v1alpha1/kamelet_binding_types.go]: fixed

**Comment by lburgazzoli** [comment]
> Should the CRD be regenerated?

I can but on my side it does re-format each yaml ... maybe someone of you can do it once merged while I'm trying to understand what is the reason

**Pull request merged**